includes/wikia/services/DataMartService.class.php

```diff
@@ -853,6 +854,45 @@ public static function getWAM200Wikis() {
 		return $wikis;
 	}
 
+	/**
+	 * Returns an array of IDs of wikias ordered by WAM rank.
+	 * The default limit is 200. If 0 is provided - no limit is used.
+	 * @param int $limit Default is set as the DEFAULT_TOP_WIKIAS_LIMIT const
+	 * @param array $wikisIds Limit the range to certain wikis
+	 * @return bool|array An array of IDs or `false` on no results
+	 */
+	public function getWikisOrderByWam( $limit = self::DEFAULT_TOP_WIKIAS_LIMIT, array $wikisIds = [] ) {
+		$app = F::app();
+
+		$db = wfGetDB( DB_SLAVE, [], $app->wg->DWStatsDB );
+
+		$sql = ( new WikiaSQL() )->skipIf( self::isDisabled() )
+			->cacheGlobal( self::TTL )
+			->SELECT( 'wiki_id' )
+			->FROM( 'dimension_top_wikis' )
+			->ORDER_BY( 'rank' );
+
+		/**
+		 * Limit the results to certain wikias
+		 */
+		if ( !empty( $wikisIds ) ) {
+			$sql->WHERE( 'wiki_id' )->IN( $wikisIds );
```

Comment:
But in our case we don't have control on this array because it's filled by ```WikiFactory::getCityIDsFromVarValue``` so we can get thousands of ids here.